cli/dispatch_minimal.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+// Do not add flatbuffer dependencies to this module.
+//! Connects to js/dispatch_minimal.ts sendAsyncMinimal This acts as a faster
+//! alternative to flatbuffers using a very simple list of int32s to lay out
+//! messages. The first i32 is used to determine if a message a flatbuffer
+//! message or a "minimal" message, see `has_minimal_token()`.
+use crate::state::ThreadSafeState;
+use deno::Buf;
+use deno::Op;
+use deno::PinnedBuf;
+use futures::Future;
+
+const DISPATCH_MINIMAL_TOKEN: i32 = 0xCAFE;
+const OP_READ: i32 = 1;
+const OP_WRITE: i32 = 2;
+
+pub fn has_minimal_token(s: &[i32]) -> bool {
+  s[0] == DISPATCH_MINIMAL_TOKEN
+}
+
+#[derive(Clone, Debug, PartialEq)]
+struct Record {
+  pub promise_id: i32,
+  pub op_id: i32,
+  pub arg: i32,
+  pub result: i32,
+}
+
+impl Into<Buf> for Record {
+  fn into(self) -> Buf {
+    let vec = vec![
+      DISPATCH_MINIMAL_TOKEN,
+      self.promise_id,
+      self.op_id,
+      self.arg,
+      self.result,
+    ];
+    //let len = vec.len();
+    let buf32 = vec.into_boxed_slice();
+    let ptr = Box::into_raw(buf32) as *mut [u8; 5 * 4];
+    unsafe { Box::from_raw(ptr) }
```

Comment:
I'm surprised that this works at all. It seems that this would create a Box<[u8; 20]> which is not the same thing as a Box<[u8]> since the latter carries length information whereas the former doesn't.
Or is there some implicit unsizing coercion going on? Questions, questions...

Reply:
i guess cause it's a pointer you can coerce it into anything

Reply:
I think I was wrong about how the type system treats slices. 
I was under the impression that `[u8]` was a fat pointer.
However it seems that `&[u8]` is a fat pointer and `[u8]` just refers to a sequence of u8 values of unspecified length. 
So I take this back, and https://github.com/denoland/deno/pull/2259#discussion_r280619229 as well.